packages/orchestration/src/orchestration-client.ts

```diff
@@ -1,63 +1,177 @@
+import { OrchestrationChatCompletionStream } from './orchestration-chat-completion-stream.js';
+import { OrchestrationChatCompletionStreamResponse } from './orchestration-chat-completion-stream-response.js';
+import type {
+  HttpResponse,
+  CustomRequestConfig
+} from '@sap-cloud-sdk/http-client';
 import type { ResourceGroupConfig } from '@sap-ai-sdk/ai-api/internal.js';
 import type { CompletionPostRequest } from './client/api/schema/index.js';
 import type {
+  LlmModuleConfig,
   OrchestrationModuleConfig,
   Prompt
 } from './orchestration-types.js';
+import type { OrchestrationChatCompletionStreamChunkResponse } from './orchestration-chat-completion-stream-chunk-response.js';
 import type { HttpDestinationOrFetchOptions } from '@sap-cloud-sdk/connectivity';
 
+interface RequestOptions {
+  prompt?: Prompt;
+  requestConfig?: CustomRequestConfig;
+  streaming?: boolean;
+  jsonConfig?: string;
+  deploymentConfig?: ResourceGroupConfig;
+  destination?: HttpDestinationOrFetchOptions;
+  config?: OrchestrationModuleConfig;
+}
+
 /**
  * Get the orchestration client.
  */
 export class OrchestrationClient {
-  /**
-   * Creates an instance of the orchestration client.
-   * @param config - Orchestration module configuration. This can either be an `OrchestrationModuleConfig` object or a JSON string obtained from AI Launchpad.
-   * @param deploymentConfig - Deployment configuration.
-   * @param destination - The destination to use for the request.
-   */
-  constructor(
-    private config: OrchestrationModuleConfig | string,
-    private deploymentConfig?: ResourceGroupConfig,
-    private destination?: HttpDestinationOrFetchOptions
-  ) {}
-
-  /**
-   * Creates a completion for the chat messages.
-   * @param prompt - Prompt configuration.
-   * @param requestConfig - Request configuration.
-   * @returns The completion result.
-   */
-  async chatCompletion(
+  static async chatCompletionWithJson(
+    jsonConfig: string,
     prompt?: Prompt,
+    deploymentConfig?: ResourceGroupConfig,
+    destination?: HttpDestinationOrFetchOptions,
     requestConfig?: CustomRequestConfig
   ): Promise<OrchestrationResponse> {
-    const body =
-      typeof this.config === 'string'
-        ? constructCompletionPostRequestFromJson(this.config, prompt)
-        : constructCompletionPostRequest(this.config, prompt);
+    const response = await OrchestrationClient.executeRequest({
+      prompt,
+      requestConfig,
+      jsonConfig,
+      deploymentConfig,
+      destination
+    });
+    return new OrchestrationResponse(response);
+  }
+
+  static async streamWithJson(
+    jsonConfig: string,
+    prompt?: Prompt,
+    controller = new AbortController(),
+    deploymentConfig?: ResourceGroupConfig,
+    destination?: HttpDestinationOrFetchOptions,
+    requestConfig?: CustomRequestConfig
+  ): Promise<
+    OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>
```

Comment:
`OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>`

Is way too long